src/transactions/simulation/TxSimUtils.cpp

```diff
@@ -26,7 +27,8 @@ replaceIssuer(Asset& asset, uint32_t partition)
         mutateScaledAccountID(asset.alphaNum12().issuer, partition);
         break;
     case ASSET_TYPE_NATIVE:
-        // nothing to do for native assets
+    case ASSET_TYPE_POOL_SHARE:
```

Comment:
I don't think this handles `ChangeTrustAsset` correctly (see line 374).

Reply:
Good catch. Fixed.